### PR TITLE
[Promotion] implement `ToggleableTrait`, fixes #3871

### LINKED ISSUE #2739: [Promotion] Applied to orders after adding
Our custom implementation of the PromotionBundle separates adding the Rules and Actions from adding a Promotion. We first add a Promotion, and thereafter add the Rules and Actions. 

This can result in an empty promotion that will be applied to an order by the processor. The documentation already states that a `Promotion` without an `Action` is not effective, but I think this should be solved in the `PromotionProcessor` directly:

1. By counting the number of Actions and not process the Promotion any further if there are no actions
2. By introducing an 'active' flag, and let the `PromotionRepository` take that flag into account.

The second option has my :+1: because otherwise the same will occur between adding two actions.

### PR DESCRIPTION
**Pull request by rvanlaak**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Related tickets | fixes #2739 , #3098 |
| License | MIT |

Implement `ToggleableTrait` on `Promotion`, also see https://github.com/Sylius/Sylius/pull/3871#issuecomment-179804903 of @pjedrzejewski

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Do you think you could add a Behat scenario for that? Would require only adding a new step saying if promotion is enabled and rest should be just matter of using existing steps.

**Comment by rvanlaak** [comment]
@pjedrzejewski added another commit for the Behat scenarios. Given this was my first Behat experience, please let me know if it makes any sense.

File: src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/PromotionRepository.php [hunk 1]
```diff
@@ -80,6 +80,7 @@ protected function filterByActive(QueryBuilder $queryBuilder, \Datetime $date =
                     $queryBuilder->expr()->gt($this->getPropertyName('endsAt'), ':date')
                 )
             )
+            ->andWhere($this->getPropertyName('enabled') . ' = 1')
```
Comment (lchrusciel) [inline src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/PromotionRepository.php]: `'o.enabled = 1'`

File: features/promotion/managing_coupons/coupon_generate_validation.feature [hunk 2]
```diff
@@ -8,6 +8,7 @@ Feature: Coupon generate instruction validation
         Given the store operates on a single channel in "France"
         And there is a promotion "Christmas sale"
         And it is coupon based promotion
+        And it is enabled promotion
```
Comment (lchrusciel) [inline features/promotion/managing_coupons/coupon_generate_validation.feature]: Promotion is enabled by default, so there is no point in writing it explicit in every scenario. This steps should be removed.

**Comment by lchrusciel** [comment]
You have missed migration file.

And behats. Can you write a new scenarios? Your PR introduce a new feature, so you should add the 'toggling_promotion.feature'. You can check [currency toggling](https://github.com/Sylius/Sylius/blob/master/features/currency/managing_currency/toggling_currency.feature) as a sample implementation. Also, you can find [this section](http://docs.sylius.org/en/latest/behat/index.html?highlight=behat) of our documentation useful.

**Comment by pamil** [comment]
@rvanlaak it would be great if you share your experiences with creating Behat scenarios (which are *quite* customized in Sylius) afterwards, I'm very interested in how the first impression looks like :)

File: src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/PromotionRepository.php (hunk 1 above)
Comment (rvanlaak) [inline src/Sylius/Bundle/PromotionBundle/Doctrine/ORM/PromotionRepository.php]: That was my first idea too, until I saw that all other fields use `getPropertyName` too.

**Comment by rvanlaak** [comment]
My first experience with Behat went like this:

> Okay, I need to add a Behat scenario, lets see how far the existing Sylius scenarios bring me. I've heard about Behat before and know about BDD, can't differ that much from PHPUnit tests. But hey wait, I have no idea how to determine what a proper Behat scenario _should_ look like. Does it has similar rules like PHPUnit has with the "you at least need two assertions for each test"? Lets dive into the test suite a bit more, and find a parameter similar to `enabled` that has tests too. So, then once again I found out that PHPStorm is awesome because it allows you to CTRL+click Behat steps and points you to the right context with the related annotation directly. This clarified a lot to me, so based on that I created the extra steps and some sort of "validated" them by assuming that if they are clickable in PHPStorm that would mean that the added Behat step would work. The last thing worth mentioning is that it takes Travis quite some time to complete the Behat test suite, so maybe I need to make a local test environment for Behat the next time too.

**Comment by Niiko** [comment]
Referring to #4496 it would be better to set enabled = true instead of enabled = 1

**Comment by rvanlaak** [comment]
@Niiko I've changed it to `enabled = true`. :+1: Can someone else fix the Behat scenarios, I'm not that known to it and next to that do not really understand what you guys expect.

**Pull request closed without merging**